githubutil/githubutil.go

```diff
@@ -440,3 +444,106 @@ const (
 	TreeModeSubmodule  = "160000"
 	TreeModeGitlink    = "120000"
 )
+
+// CreateBranch creates a branch named branchName from baseRef (e.g., "main").
+// If the branch already exists, returns ErrBranchExists.
+func CreateBranch(ctx context.Context, client *github.Client, owner, repo, branchName, baseRef string) error {
+	// Look up the base ref SHA
+	var base *github.Reference
+	if err := Retry(func() error {
+		ref, _, err := client.Git.GetRef(ctx, owner, repo, "refs/heads/"+baseRef)
+		if err != nil {
+			return err
+		}
+		base = ref
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to get base ref %s: %w", baseRef, err)
+	}
+
+	// Try to create the new ref
+	newRef := &github.Reference{
+		Ref: github.String("refs/heads/" + branchName),
+		Object: &github.GitObject{
+			SHA: base.Object.SHA,
+		},
+	}
+
+	if err := Retry(func() error {
+		_, _, err := client.Git.CreateRef(ctx, owner, repo, newRef)
+		if err != nil {
+			var errResp *github.ErrorResponse
+			if errors.As(err, &errResp) && errResp.Response != nil && errResp.Response.StatusCode == http.StatusUnprocessableEntity {
+				// This usually means the ref exists already
+				return ErrBranchExists
+			}
+			return err
+		}
+		return nil
+	}); err != nil {
+		if errors.Is(err, ErrBranchExists) {
+			return ErrBranchExists
+		}
+		return fmt.Errorf("failed to create branch %s from %s: %w", branchName, baseRef, err)
+	}
+
+	return nil
+}
+
+// CreatePullRequest creates a pull request from head to base in the given repo.
+//   - head can be either "branch" (same repo) or "owner:branch" (cross-repo).
+//   - If an open PR already exists matching the same head/base, the existing PR is returned
+//     alongside an ErrPRExists error (use errors.Is(err, ErrPRExists) to detect).
+func CreatePullRequest(
```

Comment:
I think you've re-implemented a lot of code that we already have. Take a look at https://github.com/microsoft/go-infra/blob/main/gitpr/gitpr.go